File: uber/src/main/java/com/nwtkts/uber/dto/ReportResponse.java
package com.nwtkts.uber.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ReportResponse {

    private List<ReportResponseForDay> list;

    private Double cumulativeSumOfNumberOfRides;
    private Double cumulativeSumOfPrice;
    private Double cumulativeSumOfDistance;

    private Double averageNumberOfRides;
    private Double averagePrice;
    private Double averageDistance;

}

File: uber-front/src/app/private/models/ReportResponse.ts
import { ReportResponseForDay } from "./ReportResponseForDay";

export interface ReportResponse {
    list: ReportResponseForDay[];
    cumulativeSumOfNumberOfRides: number;
    cumulativeSumOfPrice: number;
    cumulativeSumOfDistance: number;
    averageNumberOfRides: number;
    averagePrice: number;
    averageDistance: number;
    }
    